File: backend/app/schemas/reporte.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict

from app.models.enums import EstadoReporteEnum


class ReporteSchema(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    animal_id: int
    fecha: datetime
    reportante_nombre: str
    comunidad_id: int | None
    descripcion: str
    foto: str | None
    latitud: float | None
    longitud: float | None
    estado: EstadoReporteEnum


class AtencionCrear(BaseModel):
    acciones_realizadas: str
    resultado: str


class AtencionSchema(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    reporte_id: int
    fecha: datetime
    responsable: str
    acciones_realizadas: str
    resultado: str

File: frontend/src/core/modelos/reporte.ts
import type { EstadoReporte } from './enums';

export interface Reporte {
  id: number;
  animal_id: number;
  fecha: string;
  reportante_nombre: string;
  comunidad_id: number | null;
  descripcion: string;
  foto: string | null;
  latitud: number | null;
  longitud: number | null;
  estado: EstadoReporte;
}

export interface AtencionCrear {
  acciones_realizadas: string;
  resultado: string;
}

export interface Atencion {
  id: number;
  reporte_id: number;
  fecha: string;
  responsable: string;
  acciones_realizadas: string;
  resultado: string;
}
